File: src/test/scala/com/github/tetrisanalyzer/board/BoardOutlineTest.scala
package com.github.tetrisanalyzer.board

import com.github.tetrisanalyzer.BaseTest
import org.junit.{Before, Test}

class BoardOutlineTest extends BaseTest {
  var board: Board = _

  @Before def setUp {
    board = Board(Array(
      "#------x#",
      "#-----xx#",
      "#-x---xx#",
      "#x-x--xx#",
      "#########"))
  }

  @Test def getOutline {
    BoardOutline(board) should be (new BoardOutline(Array(3,2,3,4,4,1,0,0), 0))
  }

  @Test def minY {
    BoardOutline(board).minY should be (0)
  }
}

File: src/test/java/com/github/tetrisanalyzer/board/BoardOutlineTest.java
package com.github.tetrisanalyzer.board;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class BoardOutlineTest {
    private Board board;

    @Before
    public void setUp() {
        board = Board.create(
                "#------x#",
                "#-----xx#",
                "#-x---xx#",
                "#x-x--xx#",
                "#########");
    }

    @Test
    public void getOutline() {
        assertEquals(new BoardOutline(0, 3,2,3,4,4,1,0,0), new BoardOutline(board));
    }

    @Test
    public void minY() {
        assertEquals(0, new BoardOutline(board).getMinY());
    }
}
